ProjectEureka/project-eureka-flutter
# PE-82 | Integrate Video with Chat

What I did:
- Added video call functions to chat screen file
- Created a button to start video chat that will send a system message on the screen that the user started a call, and the call receiver will also see the answer button
![Screenshot_4](https://user-images.githubusercontent.com/68677129/115626155-f481f200-a2b1-11eb-874e-815d930f6017.jpg)
- Renamed chat_sceen to chat_screen. Unfortunately, Github assumes I have changed the whole code now
- Move call screen and call confirmation screen to its own folder under 'screens'

How to Test:
- Sign in with any account. It is important that your account has been registered through a new sign up form, meaning, it should have your first and last names in the system
- Choose any question and go to More Details. You must see the Answer button which you should click and choose Chat. In the chat screen, you must see the name of the question owner. You can also create your own question with an account from your physical phone, and then go to chat from another account from your emulator. That will help to test the video communication.
- Click the call button. It must redirect to the Call page. You will also see the message `<first name> started the call` in the chat. After you finish the call, it must send the message `Call ended`.
- Answer button must be visible only to the call receiver.
- Answer button should display an alert message that the call is inactive if the call was ended. 
- System messages must only appear in the center of the screen.
- User can answer the call with both the Answer button in chat and the Call button on the app bar
- When the user is receiving a call, the VideoCall button must start animation indicating the ongoing call

**NOTE:**
**- PE-82 is based on PE-85 branch, so Github assumes changes in PE-85 are also changes in PE-82**
**- Chat screen listens to the answer token continuously until the user leaves the chat screen. That has been added in order to the animated video button to turn on when the user receives a call**
